.github/actions/install-dependencies/action.yml

```diff
@@ -13,6 +13,37 @@ runs:
       with:
         node-version: ${{ inputs.node-version }}
 
+
+    - name: Restore example node_modules from cache
+      uses: actions/cache/restore@v3
+      id: restore_example_node_modules
+      with:
+        key: ${{ runner.os }}-yarn-example-${{ hashFiles('./example/yarn.lock') }}
+        restore-keys: ${{ runner.os }}-yarn-example-
+        path: ${{ github.workspace }}/example/node_modules
+
+
+    - name: Install example node_modules
+      if: steps.restore_example_node_modules.outputs.cache-hit != 'true'
+      shell: bash
+      working-directory: example
+      run: |
+        # Retry 3 times before the steps actually fails
+        (echo "===== Install node_modules Attempt:  1 ====" && yarn install --frozen-lockfile) || \
+        (echo "===== Install node_modules Attempt:  2 ====" && yarn install --frozen-lockfile) || \
+        (echo "===== Install node_modules Attempt:  3 ====" && yarn install --frozen-lockfile) || \
+        (echo "===== Install node_modules Step Failed ====" && exit 1)
+
+
+    - name: Save example node_modules to cache
+      if: steps.restore_example_node_modules.outputs.cache-hit != 'true'
+      uses: actions/cache/save@v3
+      with:
+        key: ${{ steps.restore_example_node_modules.outputs.cache-primary-key }}
+        path: ${{ github.workspace }}/example/node_modules
```

Comment:
manual save/restore and installing node_modules 3 times seem unnecessary.

i'd also suggest moving this out to a composite action to reuse in multiple workflows https://github.com/callstack/react-native-builder-bob/blob/main/packages/create-react-native-library/templates/common/%24.github/actions/setup/action.yml

Reply:
We install node_modules only if `yanr.lock` changes on `cache-node-modules` step of pipeline 

the rest pipeline steps will just download cached `node_modules`

---

I will add `lookup-only: true` for install step to avoid unnecessary  download

Reply:
```
https://github.com/callstack/react-native-builder-bob/blob/main/packages/create-react-native-library/templates/common/%24.github/actions/setup/action.yml
```
p.s. using `hashFiles('**/yarn.lock')`  leads to unstability  of cache key as these files can be found inside node_modules 


to test:
```
find . -type f -name "yarn.lock" 
```

Reply:
> We install node_modules only if yanr.lock changes on cache-node-modules step of pipeline
> the rest pipeline steps will just download cached node_modules

doesn't the linked action do the same? it should download cached `node_module` when restore key matches. we always download to skip if some stuff is already there but that can be removed.

> I will add lookup-only: true for install step to avoid unnecessary download

Is it downloading if cache hit for the restore keys didn't occur? Which cache is it downloading?

For the saving cache, it's already be skipped if cache hit occurred when using `actions/cache@v3` directly.

> using hashFiles('**/yarn.lock') leads to unstability of cache key as these files can be found inside node_modules

Fair point. Bob actually doesn't need the `**/` as there's a single `yarn.lock` as it's a monorepo.

Reply:
> doesn't the linked action do a similar thing?
no
- caching & uploading archive happens only as a `post` step (in the end of pipeline)
  - so `node_modules` won't be cached if your checks (lint, tels) will fail